hack/lib/bash.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+#
+# Copyright (C) 2024 ScyllaDB
+#
+
+set -euExo pipefail
+shopt -s inherit_errexit
+
+function cleanup-bg-jobs() {
+  ec=$1
+  mapfile -t bg_jobs < <( jobs -p )
+  if [[ ${#bg_jobs[@]} -ne 0 ]]; then
+    printf "Found %d leftover jobs" >> /dev/stderr
+    jobs
+    printf "Killing %d leftover jobs" "${#bg_jobs[@]}" >> /dev/stderr
+    kill "${bg_jobs[@]}"
+    echo "Waiting for terminating jobs to finish" >> /dev/stderr
+    wait
+    exit $(( 127 + ec ))
```

Comment:
I don't understand returning >=127 here, would you mind explaining?

Reply:
it a linux practice to distinquish failures from the initial command combined with someting else that failed